src/applications/vaos/api/index.js

```diff
@@ -169,29 +169,33 @@ export function getSystemDetails(systemIds) {
   }
 
   return promise.then(resp =>
-    resp.data
-      .map(item => ({ ...item.attributes, id: item.id }))
-      // Sometimes facilities that aren't in our codes list come back, because they're
-      // marked as parents. We don't want this, so we're filtering them out
-      .filter(item => item.rootStationCode === item.institutionCode),
+    resp.data.map(item => ({ ...item.attributes, id: item.id })),
   );
 }
 
-export function getFacilitiesBySystemAndTypeOfCare(systemId, typeOfCareId) {
+export function getFacilitiesBySystemAndTypeOfCare(
+  systemId,
+  parentId,
+  typeOfCareId,
+) {
   let promise;
   if (USE_MOCK_DATA) {
-    if (systemId === '984') {
+    if (parentId === '984') {
       promise = import('./facilities_984.json').then(
         module => (module.default ? module.default : module),
       );
+    } else if (parentId === '983A6') {
+      promise = import('./facilities_983A6.json').then(
+        module => (module.default ? module.default : module),
+      );
     } else {
       promise = import('./facilities_983.json').then(
         module => (module.default ? module.default : module),
       );
     }
   } else {
     promise = apiRequest(
-      `/vaos/systems/${systemId}/direct_scheduling_facilities?type_of_care_id=${typeOfCareId}`,
+      `/vaos/systems/${systemId}/direct_scheduling_facilities?type_of_care_id=${typeOfCareId}&parent_code=${parentId}`,
```

Comment:
Important change adding the parameter back in to restrict the facilities here to direct children of `parent_cod`, not just `systemId`.